evidence/verify.go

```diff
@@ -55,64 +76,100 @@ func (evpool *Pool) verify(evidence types.Evidence) error {
 		if err != nil {
 			return err
 		}
-		return VerifyDuplicateVote(ev, state.ChainID, valSet)
+
+		if err := VerifyDuplicateVote(ev, state.ChainID, valSet); err != nil {
+			return types.NewErrInvalidEvidence(evidence, err)
+		}
+
+		return nil
 
 	case *types.LightClientAttackEvidence:
 		commonHeader, err := getSignedHeader(evpool.blockStore, evidence.Height())
 		if err != nil {
-			return err
+			return types.NewErrInvalidEvidence(evidence, err)
```

Comment:
This is just trying to retrieve a signed header from the store so we should just return a normal error instead of an invalid evidence one

Reply:
Nice catch -- updated.